server/game/cards/06_SEC/units/HiredSlicer.ts

```diff
@@ -0,0 +1,61 @@
+import type { IAbilityHelper } from '../../../AbilityHelper';
+import type { INonLeaderUnitAbilityRegistrar } from '../../../core/card/AbilityRegistrationInterfaces';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { TargetMode, WildcardCardType } from '../../../core/Constants';
+
+export default class HiredSlicer extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '1504187296',
+            internalName: 'hired-slicer'
+        };
+    }
+
+    public override setupCardAbilities(registrar: INonLeaderUnitAbilityRegistrar, AbilityHelper: IAbilityHelper) {
+        registrar.addOnAttackAbility({
+            title: 'Reveal the top 2 cards of a deck. If you do, exhaust a unit that shares a trait with one of those cards',
+            optional: true,
+            // Player chooses which deck to reveal from
+            targetResolver: {
+                mode: TargetMode.Select,
+                choices: (context) => ({
+                    'Your deck': AbilityHelper.immediateEffects.reveal(() => ({
+                        target: context.player.getTopCardsOfDeck(2)
+                    })),
+                    'Opponent\'s deck': AbilityHelper.immediateEffects.reveal(() => ({
+                        target: context.player.opponent.getTopCardsOfDeck(2)
+                    }))
```

Comment:
Nit: I think you can pass in the props directly instead of using a generator function with empty parameter list. So basically, each reveal effect would look like:

```typescript
AbilityHelper.immediateEffects.reveal({
    target: context.player.opponent.getTopCardsOfDeck(2)
})
```